internal/api/installation_db_migration.go

```diff
@@ -0,0 +1,328 @@
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	outputJSON(c, w, dbMigrationOperation)
+}
+
+// handleRollbackInstallationDatabaseMigration responds to POST /api/installations/operations/database/migration/{migration}/rollback,
+// rollbacks database migration.
+func handleRollbackInstallationDatabaseMigration(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	migrationID := vars["migration"]
+
+	c.Logger = c.Logger.WithField("action", "rollback-installation-database-migration").
+		WithField("migration-operation", migrationID)
+
+	dbMigrationOperation, status, unlockOnce := lockInstallationDBMigrationOperation(c, migrationID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
+	newState := model.InstallationDBMigrationStateRollbackRequested
+
+	if !dbMigrationOperation.ValidTransitionState(newState) {
+		c.Logger.Warn("Cannot rollback migration, invalid state")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	installationDTO, status, unlockInstOnce := lockInstallation(c, dbMigrationOperation.InstallationID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockInstOnce()
+
+	if installationDTO.State != model.InstallationStateHibernating {
+		c.Logger.Error("Installation needs to be hibernated to be rolled back")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	err := c.Store.TriggerInstallationDBMigrationRollback(dbMigrationOperation, installationDTO.Installation)
+	if err != nil {
+		c.Logger.WithError(err).Error("Failed to trigger db migration rollback")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	unlockOnce()
+	c.Supervisor.Do()
+
+	w.WriteHeader(http.StatusAccepted)
+	outputJSON(c, w, dbMigrationOperation)
+}
+
+func validateDBMigration(c *Context, installation *model.Installation, migrationRequest *model.InstallationDBMigrationRequest, currentDB *model.MultitenantDatabase) error {
+	if migrationRequest.DestinationDatabase != model.InstallationDatabaseMultiTenantRDSPostgres ||
+		installation.Database != model.InstallationDatabaseMultiTenantRDSPostgres {
+		return errors.Errorf("db migration is supported when both source and destination are %q database", model.InstallationDatabaseMultiTenantRDSPostgres)
+	}
+
+	if migrationRequest.DestinationMultiTenant == nil {
+		return errors.New("destination database data not provided")
+	}
+
+	destinationDB, err := c.Store.GetMultitenantDatabase(migrationRequest.DestinationMultiTenant.DatabaseID)
+	if err != nil {
+		return errors.Wrap(err, "failed to get destination multi-tenant database")
+	}
+	if destinationDB == nil {
+		return errors.Errorf("destination database with id %q not found", migrationRequest.DestinationMultiTenant.DatabaseID)
+	}
+	if destinationDB.DatabaseType != model.DatabaseEngineTypePostgres {
+		return errors.Errorf("destination database is not Postgres")
+	}
```

Comment:
Isn't this redundant with the earlier check for `InstallationDatabaseMultiTenantRDSPostgres`?